toolkit/tools/pkg/imagecustomizerlib/artifactsinputoutput.go

```diff
@@ -24,47 +26,76 @@ const (
 	SystemdBootDir = "EFI/systemd"
 )
 
+type OutputArtifactMetadata struct {
+	Partition   InjectFilePartition `yaml:"partition"`
+	Destination string              `yaml:"destination"`
+	Source      string              `yaml:"source"`
+}
+
+type InjectFilePartition struct {
+	MountIdType imagecustomizerapi.MountIdentifierType `yaml:"mountIdType"`
+	Id          string                                 `yaml:"id"`
+}
+
+type InjectFilesYaml struct {
+	InjectFiles []OutputArtifactMetadata `yaml:"injectFiles"`
+}
+
 var ukiRegex = regexp.MustCompile(`^vmlinuz-.*\.efi$`)
 
-func outputArtifacts(items []imagecustomizerapi.OutputArtifactsItemType, outputDir string, buildDir string, buildImage string) error {
+func outputArtifacts(items []imagecustomizerapi.OutputArtifactsItemType,
+	configOutputArtifactsPath string, buildDir string, buildImage string, baseConfigPath string,
+) ([]OutputArtifactMetadata, error) {
 	logger.Log.Infof("Outputting artifacts")
 
+	var outputArtifactsMetadata []OutputArtifactMetadata
+	outputDir := file.GetAbsPathWithBase(baseConfigPath, configOutputArtifactsPath)
+	prefix := ""
+	if !filepath.IsAbs(configOutputArtifactsPath) {
+		prefix = "./"
```

Comment:
I think the metadata file should be added to the `configOutputArtifactsPath` directory. Also, all the paths in the metadata file should be relative, regardless of if `configOutputArtifactsPath` is relative or not.

Reply:
Revised, please have a check.